src/middlewares/superadminAuth.js

```diff
@@ -0,0 +1,30 @@
+import jwt from 'jsonwebtoken';
+
+class superadminAuth {
+  static verifyToken = async (req, res, next) => {
+    const token = req.header('Authorization').replace('Bearer ', '');
+
+    if (!token) {
+      return res
+        .status(403)
+        .json({ message: 'Token Not Provided Please Check Headers' });
+    }
+
```

Comment:
@itisWasp  This is already implemented in another middleware.
Just call that middleware before this superAdminAuth in your route.
Before checking is someone is a superAdmin (role)  he must be authenticated first